ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/StateProcessor.java

```diff
@@ -83,9 +83,9 @@ public StateProcessor(ServiceConfig config, ChainStorageClient chainStorageClien
     this.eventBus = config.getEventBus();
     this.config = config.getConfig();
     this.publicKey = config.getKeyPair().publicKey();
-    this.beaconChainStateMetrics = new BeaconChainStateMetrics(config.getMetricsSystem());
-    this.eventBus.register(this);
+    this.stateTransition = new StateTransition(true, new EpochMetrics(config.getMetricsSystem()));
```

Comment:
ok so we are back to instantiating stateTransition once in the StateProcessor instead of each time ForkChoiceUtil is called?  I like that a lot more.